packages/runtime/runtime-utils/src/dataStoreHelpers.ts

```diff
@@ -36,40 +55,24 @@ export function getStack() {
 }
 
 export function exceptionToResponse(err: any): IResponse {
-    const status = 500;
-    // eslint-disable-next-line no-null/no-null
-    if (err !== null && typeof err === "object" && err.errorFromRequestFluidObject === true) {
-        const responseErr: IResponseException = err;
+    if (ResponseException.is(err)) {
         return {
             mimeType: "text/plain",
-            status: responseErr.code,
```

Comment:
We are losing here the most important thing you are adding above - requestRoute property.
It feels like instead of having various disjoined fields, we should have one field of type ResponseException here and populated it in createResponseError.

Also note that I missed it (and was wrong in my prior assessment) that we do capture URI in createResponseError() and it escapes your new tagging. So there is more reason to have unified structure (i.e. using and populating ResponseException from the start and preserving it across all these calls).

I actually like it for different reason - it's unfortunate that we chose to invent alternative way to return errors when language already provides one. So the more we make it really thin wrapper around standard way of error handling (using error object and assume it's the one thrown and preserved through transitions), the better.

Reply:
Would it be going too far to add a type IFluidRoutingError to IFluidObject and return that in error cases? I remember discussing always using IFluidObject as the type for `value` and retiring mimeType last year, so that's what brought this to mind.

Reply:
Also, I was feeling bad bit having written any tests... seems more necessary as we discuss more...